docs/en/observability/monitor-gcp.asciidoc

```diff
@@ -0,0 +1,457 @@
+Store the Cloud ID in the keystore.
++
+[source,bash]
+----
+./metricbeat keystore create
+echo -n "<Your Deployment Cloud ID>" | ./metricbeat keystore add CLOUD_ID --stdin
+----
+
+. To store metrics in {es} with minimal permissions, create an API key to send
+data from {metricbeat} to {ess}. Log into Kibana (you can do so from the Cloud
+Console without typing in any permissions) and select *Management* -> *Dev
+Tools*. Then, send the following request:
++
+[source,console]
+----
+POST /_security/api_key
+{
+  "name": "metricbeat-monitor-gcp",
+  "role_descriptors": {
+    "metricbeat_writer": {
+      "cluster": ["monitor", "read_ilm"],
+      "index": [
+        {
+          "names": ["metricbeat-*"],
+          "privileges": ["view_index_metadata", "create_doc"]
+        }
+      ]
+    }
+  }
+}
+----
+
+. The response contains an `api_key` and an `id` field, which can be stored in
+the {metricbeat} keystore in the following format: `id:api_key`.
++
+[source,bash]
+----
+echo -n "IhrJJHMB4JmIUAPLuM35:1GbfxhkMT8COBB4JWY3pvQ" | ./metricbeat keystore add ES_API_KEY --stdin
+----
++
+[NOTE]
+=====
+Make sure you specify the `-n` parameter; otherwise, you will have
+painful debugging sessions, due to adding a newline at the end of
+your API key.
+=====
+
+. To see if both settings have been stored, run `./metricbeat keystore list`.
+
+. Then, configure {metricbeat} to output to {ess}. Edit the `metricbeat.yml`
+configuration file and add the following lines to the end of the file.
++
+[source,yml]
+----
+cloud.id: ${CLOUD_ID}
+output.elasticsearch:
+  api_key: ${ES_API_KEY}
+----
+
+. Finally, test if the configuration is working. If it is not working,
+verify if you used the right credentials and add them again.
++
+[source,bash]
+----
+./metricbeat test output
+----
+
+Now that the output is working you are going to setup the input (GCP).
+
+[discrete]
+=== Step 3: Configure {metricbeat} Google Cloud Platform module
+
+In order to collect metrics from Google Cloud Platform, use the
+https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-gcp.html[Google Cloud Platform]
+module. This module periodically fetches monitoring metrics from Google Cloud
+Platform using
+https://cloud.google.com/monitoring/api/metrics_gcp[Stackdriver Monitoring API]
+for Google Cloud Platform services.
+
+[WARNING]
+====
+Extra GCP charges on Stackdriver Monitoring API requests may be generated by
+this module. Please see
+https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-gcp.html#gcp-api-requests[rough estimation of the number of API calls]
+for more details.
+====
+
+. First, enable the GCP module.
++
+[source,bash]
+----
+./metricbeat modules enable gcp
+----
+
+. Next, edit the `modules.d/gcp.yml` file to configure which metrics to
+collect.
++
+[source,yml]
+----
+- module: googlecloud
+  metricsets:
+    - compute <1>
+  zone: "" <2>
+  project_id: "elastic-education" <3>
```

Comment:
you may want to change that id, as it is not something readers can identify with?